File: trimming-middle/500/data/wordcount/src/wordcount.go
package main

import (
	"fmt"
	"strings"
)

// 単語数を数える関数
func countWords(text string) int {
	// 空白で分割してスライスにする
	words := strings.Fields(text)
	return len(words)
}

func main() {
	text := "WebAssembly is a binary instruction format for a stack-based virtual machine"
	
	fmt.Println("Text:", text)
	
	count := countWords(text)
	fmt.Printf("Word Count: %d\n", count)
}

File: trimming-head/1000/data/wordcount/src/wordcount.c
#include <stdio.h>
#include <ctype.h>
#include <stdbool.h>

// 単語数を数える関数
int countWords(const char *text) {
    int count = 0;
    bool inWord = false;

    for (const char *p = text; *p; p++) {
        // 空白文字かどうかを判定
        if (isspace(*p)) {
            inWord = false;
        } else if (!inWord) {
            // 空白から文字に変わった瞬間をカウント
            inWord = true;
            count++;
        }
    }
    return count;
}

int main() {
    const char *text = "WebAssembly is a binary instruction format for a stack-based virtual machine";
    
    printf("Text: %s\n", text);
    
    int count = countWords(text);
    printf("Word Count: %d\n", count);

    return 0;
}